### PR TITLE
Item: #0000 Desc: Add port to python3 of template_engine

### PR DESCRIPTION
**Pull request by erzetpe**

### CODE REVIEW

File: core/src/core/template_engine_3 [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        # Will only produce a dump of the data to stdout
+        if not options.data_file:
+            p.print_help()
+            sys.exit()
+    else:
+        if not options.input_file or not options.output_file or not options.data_file:
+            p.print_help()
+            sys.exit()
+
+    options_display = ""
```
Comment (lukurde) [inline core/src/core/template_engine_3]: Is this variable used anywhere?

File: core/src/core/template_engine_3 [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+    logging.basicConfig(level=logging.INFO)
+    log = logging.getLogger(__name__)
+
+    # If the -j flag was passed then convert the yaml to pretty json in sorted order
+    if json_arg:
+        with open(data_file) as data:
+            template_dict = yaml.load(data)
+        log.info(json.dumps(template_dict, indent=4, sort_keys=True))
+        exit(0)
```
Comment (lukurde) [inline core/src/core/template_engine_3]:
```suggestion
        sys.exit(0)
```
see: https://stackoverflow.com/questions/6501121/difference-between-exit-and-sys-exit-in-python

File: core/src/core/template_engine_3 [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+    if yaml_arg:
+        with open(data_file) as data:
+            template_dict = json.load(data)
```
[the remark this answers is not in the document]
Comment (erzetpe) [reply core/src/core/template_engine_3]: This works as intended. Take json and converts to yaml and take json and converts to yaml.

File: core/src/core/template_engine_3 (hunk 1 above)
Comment (erzetpe) [reply core/src/core/template_engine_3]: No. My mistake. :)

File: core/src/core/template_engine_3 (hunk 2 above)
Comment (erzetpe) [reply core/src/core/template_engine_3]: Changed according to documentation.

File: core/src/core/template_engine_3 [hunk 4]
```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (erzetpe) [reply core/src/core/template_engine_3]: Changed according to suggestion.

File: core/src/core/template_engine_3 [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+import yaml
+from jinja2 import Environment, FileSystemLoader
+
+_SEPARATOR = '=' * 60
+
+
+# All three file paths must be full paths to each.
+def render_template(data_file, in_file, out_file, json_arg, yaml_arg):
+    template_dict = ""
```
[the remark this answers is not in the document]
Comment (erzetpe) [reply core/src/core/template_engine_3]: Changed according to suggestion. Fixed earlier code.

File: core/src/core/template_engine_3 (hunk 2 above)
Comment (erzetpe) [reply core/src/core/template_engine_3]: Changed according to suggestion.

**Pull request merged**